core/src/androidMain/kotlin/Peripheral.kt

```diff
@@ -78,34 +77,39 @@ public class AndroidPeripheral internal constructor(
         combine(_ready, state) { ready, state -> ready && state == State.Connected }.first { it }
     }
 
+    private fun establishConnection(): Connection =
+        bluetoothDevice.connect(
+            context,
+            _state,
+            invokeOnClose = { connectJob.value = null }
+        ) ?: throw ConnectionRejectedException()
+
+    /** Creates a connect [Job] that completes when connection is established, or failure occurs. */
     private fun createConnectJob(): Job = scope.launch(start = LAZY) {
         _ready.value = false
 
-        val connection =
-            bluetoothDevice.connect(context, _state) ?: throw ConnectionRejectedException()
-        _connection = connection
+        val connection = establishConnection().also { _connection = it }
```

Comment:
Was the switch to using "also" here stylistic? Or is there a more concrete reason to do this instead of just assigning _connection?

Reply:
The `establishConnection` call shortened the line a bit, so make it so it was still reasonably readable when using `also`.

**tl;dr** yes, pure stylistic